### PR TITLE
Rewards: Reset the verified publisher badge when visiting a local URL

### LINKED ISSUE #195: Verified publisher badge remains visible when tab is closed (New Tab UI)
The verified label should be updated when closing a tab

Simulator Screen Shot - iPhone Xs - 2019-08-28 at 09 56 56

### PR DESCRIPTION
**Pull request by danishjafri88**
Fixes brave/brave-rewards-ios#195

## Summary of Changes

This pull request fixes issue: Fixes: brave/brave-rewards-ios#195 by resetting the verified state on panel fetch mismatch data
Fix on brave-core side: https://github.com/brave/brave-core/pull/3404

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Test Plan:

-> visit a site that is verified and the verified symbol is shown.
-> open a new tab and check the URL bar for the verified symbol

Expected:
No verified symbol should be displayed.

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/internal/issues/new?assignees=&labels=security%2C+security-reviews&template=security-review.md&title=iOS:%20Security+Review+for) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 1]
```diff
@@ -230,9 +230,12 @@ class BrowserViewController: UIViewController {
         guard let rewards = rewards, let observer = rewardsObserver else { return }
         rewards.ledger.add(observer)
         observer.fetchedPanelPublisher = { [weak self] publisher, tabId in
-            guard let self = self, self.isViewLoaded, let tab = self.tabManager.selectedTab, tab.rewardsId == tabId else { return }
-            self.publisher = publisher
-            self.updateRewardsButtonState()
+            guard let strongSelf = self, strongSelf.isViewLoaded, let tab = strongSelf.tabManager.selectedTab, tab.rewardsId == tabId else {
+                self?.resetRewardsButtonVerifiedState()
```
Comment (kylehickinson) [inline Client/Frontend/Browser/BrowserViewController.swift]: I don't think this solves the underlying issue, which is that when new tabs are being shown, the `fetchedPanelPublisher` observer method is reporting the old tab as the fetched publisher. This is possibly a bug in brave-core side of things (`onTabRetrieved`/reporting methods in BATBraveLedger.mm)

Steps to see this: 
1. Open a tab, go to amazon.com
2. Breakpoint on line 233 (guard statement)
3. Open a new tab, breaks again but the tabId being shown is still amazon.com's tab, and should be the new tab (Can be checked by dumping `rewardsId` on tabManager.allTabs)

**Review by kylehickinson — CHANGES_REQUESTED** [review]

**Review by kylehickinson — COMMENTED** [review]
Does this PR not require the changes made in brave-core?

**Comment by danishjafri88** [comment]
> Does this PR not require the changes made in brave-core?

The core pr makes sure no callback is done on unselected tabs/invalid urls
This one removes the verified mark on new tab(homepage or any other local url)
They are independent of each other

**Comment by kylehickinson** [comment]
@danishjafri88 Yes I get that, but are both required to actually close the ticket as fixed?

**Comment by danishjafri88** [comment]
Yup

**Comment by kylehickinson** [comment]
Okay, so then we'll want to update the `brave-rewards-ios` library while checked into the ios-pre-v3-fork branch (`./scripts/build_in_core.sh --skip-update /path/to/brave-browser`), push those, and then update the dependency in brave-ios (`carthage update brave-rewards-ios --platform iOS --cache-builds`)  and include the `Cartfile.resolved` update in this PR

**Comment by danishjafri88** [comment]
> Okay, so then we'll want to update the `brave-rewards-ios` library while checked into the ios-pre-v3-fork branch (`./scripts/build_in_core.sh --skip-update ~/git/brave/brave-browser`), push those, and then update the dependency in brave-ios (`carthage update brave-rewards-ios --platform iOS --cache-builds`) and include the `Cartfile.resolved` update in this PR

Yes, FYI brave-browser should point to `56f8f74fe222e928a552c45fa2b6a754e51ab88d `

**Comment by kylehickinson** [comment]
Thats fine, in brave-rewards-ios you'll end up with a README.md update that shows that brave-browser is pointing at said SHA and brave-core is pointing to `f2290c419634ea058a5d6230644ec77d19182522`

**Review by kylehickinson — APPROVED** [review]

**Pull request merged**